Libraries/xcassets/Sources/Slot/SystemVersion.cpp

```diff
@@ -6,10 +6,10 @@
  LICENSE file in the root directory of this source tree.
  */
 
-#include <xcassets/Slot/SystemVersion.h>
-
 #include <sstream>
 
+#include <xcassets/Slot/SystemVersion.h>
```

Comment:
Why is it necessary to move this include? Does the build break on BSD otherwise?

Reply:
sstream contains the macro definition of the freebsd system major, minor, mothis include SystemVersion.h to override 
SystemVersion.h
#ifdef major
#undef major
#endif 
#ifdef minor
#undef minor
#endif
build error message:
/usr/bin/c++  -Dxcassets_EXPORTS -I../Libraries/xcassets/Headers -I../Libraries/libutil/Headers -I../Libraries/ext/Headers -I../Libraries/plist/Headers -std=c++11 -fno-rtti -fno-exceptions -fPIC   -Wall -Werror -Wno-delete-non-virtual-dtor -Wno-self-assign-field -fcolor-diagnostics -MD -MT Libraries/xcassets/CMakeFiles/xcassets.dir/Sources/Slot/SystemVersion.cpp.o -MF Libraries/xcassets/CMakeFiles/xcassets.dir/Sources/Slot/SystemVersion.cpp.o.d -o Libraries/xcassets/CMakeFiles/xcassets.dir/Sources/Slot/SystemVersion.cpp.o -c ../Libraries/xcassets/Sources/Slot/SystemVersion.cpp
../Libraries/xcassets/Sources/Slot/SystemVersion.cpp:87:26: error: no member named '__major' in 'xcassets::Slot::SystemVersion'; did you mean 'major'?
    out << systemVersion.major();
                         ^
/usr/include/sys/types.h:389:18: note: expanded from macro 'major'
#define major(d)        __major(d)
                        ^
../Libraries/xcassets/Headers/xcassets/Slot/SystemVersion.h:45:9: note: 'major' declared here
    int major() const
        ^
../Libraries/xcassets/Sources/Slot/SystemVersion.cpp:89:26: error: no member named '__minor' in 'xcassets::Slot::SystemVersion'; did you mean 'minor'?
    out << systemVersion.minor();
                         ^
/usr/include/sys/types.h:395:18: note: expanded from macro 'minor'
#define minor(d)        __minor(d)
                        ^
../Libraries/xcassets/Headers/xcassets/Slot/SystemVersion.h:51:9: note: 'minor' declared here
    int minor() const
        ^
2 errors generated.

Reply:
I understand. In this case I think the proper fix is to `#include <sstream>` directly in `SystemVersion.h`. Otherwise every file that includes `SystemVersion.h` will have to do this trickery of ordering includes properly with `sstream`.

Reply:
Thanks for adding this include in `SystemVersion.h`. Can you also undo the `#include` reordering in this file?